src/library/CharacterManifestData.js

```diff
@@ -638,12 +669,21 @@ class TraitTexturesGroup{
       }
     });
   }
-  createCollection(itemCollection, replaceExisting = false){
+  createCollection(itemCollection, replaceExisting = false, ownedTraitsArray = null){
```

Comment:
Instead of always passing `ownedTraitsArray = null` to every functions, maybe it would be cleaner to ~~create a new class `"NFTTraitsManager"`~~ use the OwnedTraitIDs class and have that class provide all the information about NFT traits; 

logic would become something like
```js

//    if (ownedTraitsArray == null){
//      getAsArray(itemCollection).forEach(item => {
//        this.collection.push(new TextureTrait(this, item))
//      });
//    }
//    else{
//      getAsArray(itemCollection).forEach(item => {
//        if (ownedTraitsArray.includes(item.id)){
//          this.collection.push(new TextureTrait(this, item))
//        }
//      });
//    }



      getAsArray(itemCollection).forEach(item => {
          if( !NFTTraitsManager.isNFT(item) || NFTTraitsManager.isOwned(item) ) {
                  this.collection.push(new TextureTrait(this, item))
           }

      });

```

isOwned could just be something like 

```js

 class  OwnedTraitIDs {
 
     ....

     
       isOwned (n) {
          this.nftTraits has i = n.tokenID && has n.collection.name; idk
          }
          
      // I realize this function is redundant         
     isNFT(n){
       return this.Owned(n)
     }
          
```